neo/SmartContract/NefFile.cs

```diff
@@ -0,0 +1,121 @@
+﻿using Neo.Cryptography;
+using Neo.IO;
+using System;
+using System.IO;
+
+namespace Neo.SmartContract
+{
+    /// <summary>
+    /// +------------+-----------+------------------------------------------------------------+
+    /// |   Field    |  Length   |                          Comment                           |
+    /// +------------+-----------+------------------------------------------------------------+
+    /// | Magic      | 4 bytes   | Magic header                                               |
+    /// | Compiler   | 32 bytes  | Compiler used                                              |
+    /// | Version    | 16 bytes  | Compiler version (Mayor, Minor, Build, Version)            |
+    /// | ScriptHash | 20 bytes  | ScriptHash for the script                                  |
+    /// | Script     | Var bytes | Var bytes for the payload                                  |
+    /// | Checksum   | 4 bytes   | Sha256 of the whole file whithout the last for bytes(CRC)  |

```

Comment:
We can only compute the checksum of the header. Because `Script` has already checked by `ScriptHash`.

Reply:
Usually the checksum is for the whole file

Reply:
Maybe we should remove the script hash and move this to neo-vm?

Reply:
I've thought about it. I think vm should only need to care about script.

Reply:
So, what i need to do for merge this?

Reply:
```c#
// +------------+-----------+------------------------------------------------------------+
// |   Field    |  Length   |                          Comment                           |
// +------------+-----------+------------------------------------------------------------+
// | Magic      | 4 bytes   | Magic header                                               |
// | Compiler   | 32 bytes  | Compiler used                                              |
// | Version    | 16 bytes  | Compiler version (Mayor, Minor, Build, Version)            |
// | ScriptHash | 20 bytes  | ScriptHash for the script                                  |
// +------------+-----------+------------------------------------------------------------+
// | Checksum   | 4 bytes   | Sha256 of the header (CRC)                                 |
// +------------+-----------+------------------------------------------------------------+
// | Script     | Var bytes | Var bytes for the payload                                  |
// +------------+-----------+------------------------------------------------------------+
```

I think we should change the layout.

Reply:
Done!